### PR TITLE
X86: dts linker / memory node updates

### PR DESCRIPTION
**Pull request by galak**

### CODE REVIEW

File: arch/x86/core/x86_mmu.c [hunk 1]
```diff
@@ -15,10 +15,13 @@
...
 LOG_MODULE_DECLARE(os);
 
+#define PHYS_RAM_ADDR DT_REG_ADDR(DT_CHOSEN(zephyr_sram))
+#define PHYS_RAM_SIZE DT_REG_SIZE(DT_CHOSEN(zephyr_sram))
+
 /* Despite our use of PAE page tables, we do not (and will never) actually
  * support PAE. Use a 64-bit x86 target if you have that much RAM.
  */
-BUILD_ASSERT(DT_PHYS_RAM_ADDR + (DT_RAM_SIZE * 1024ULL) - 1ULL <=
+BUILD_ASSERT(PHYS_RAM_ADDR + PHYS_RAM_SIZE - 1ULL <=
```
Comment (andrewboie) [inline arch/x86/core/x86_mmu.c:24]: This isn't right, PHYS_RAM_SIZE needs to be multiplied by 1024 to get it in bytes not kilobytes.
Consider just incorporating the 1024 coefficient to the PHYS_RAM_SIZE definition, will make this simpler.

**Review by tyrantlizard77 — CHANGES_REQUESTED** [review]

File: arch/x86/core/x86_mmu.c (hunk 1 above)
Comment (galak) [reply arch/x86/core/x86_mmu.c:24]: PHYS_RAM_SIZE is now the size in bytes.

Comment (galak) [reply arch/x86/core/x86_mmu.c:24]: DT_REG_SIZE() gives us the size in bytes, that I removed the 1024 scaling.

I wonder if there's a common x86 header to put:
```
#define PHYS_RAM_ADDR DT_REG_ADDR(DT_CHOSEN(zephyr_sram))
#define PHYS_RAM_SIZE DT_REG_SIZE(DT_CHOSEN(zephyr_sram))
```

File: arch/x86/core/x86_mmu.c [hunk 2]
```diff
@@ -665,10 +668,10 @@ static inline bool is_within_system_ram(uintptr_t addr)
 {
 #ifdef CONFIG_X86_64
 	/* FIXME: locore not included in CONFIG_SRAM_BASE_ADDRESS */
-	return addr < (DT_PHYS_RAM_ADDR + (DT_RAM_SIZE * 1024U));
+	return addr < (PHYS_RAM_ADDR + (PHYS_RAM_SIZE * 1024U));
```
Comment (dcpleung) [inline arch/x86/core/x86_mmu.c:671]: No need to multiply by 1024 anymore.

File: arch/x86/core/x86_mmu.c [hunk 3]
```diff
@@ -665,10 +668,10 @@ static inline bool is_within_system_ram(uintptr_t addr)
...
 #else
-	return (addr >= DT_PHYS_RAM_ADDR) &&
-		(addr < (DT_PHYS_RAM_ADDR + (DT_RAM_SIZE * 1024U)));
+	return (addr >= PHYS_RAM_ADDR) &&
+		(addr < (PHYS_RAM_ADDR + (PHYS_RAM_SIZE * 1024U)));
```
Comment (dcpleung) [inline arch/x86/core/x86_mmu.c:674]: No need to multiply by 1024 anymore.

File: arch/x86/core/x86_mmu.c [hunk 4]
```diff
@@ -1069,15 +1072,15 @@ static void apply_mem_partition(struct x86_page_tables *ptables,
...
 	}
 
-	__ASSERT(partition->start >= DT_PHYS_RAM_ADDR,
+	__ASSERT(partition->start >= PHYS_RAM_ADDR,
 		 "region at %08lx[%zu] extends below system ram start 0x%08x",
-		 partition->start, partition->size, DT_PHYS_RAM_ADDR);
+		 partition->start, partition->size, PHYS_RAM_ADDR);
 	__ASSERT(((partition->start + partition->size) <=
-		  (DT_PHYS_RAM_ADDR + (DT_RAM_SIZE * 1024U))),
+		  (PHYS_RAM_ADDR + (PHYS_RAM_SIZE * 1024U))),
```
Comment (dcpleung) [inline arch/x86/core/x86_mmu.c:1079]: No need to multiply by 1024 anymore.

File: arch/x86/core/x86_mmu.c [hunk 5]
```diff
@@ -1069,15 +1072,15 @@ static void apply_mem_partition(struct x86_page_tables *ptables,
...
 		 "region at %08lx[%zu] end at %08lx extends beyond system ram end 0x%08x",
 		 partition->start, partition->size,
 		 partition->start + partition->size,
-		 (DT_PHYS_RAM_ADDR + (DT_RAM_SIZE * 1024U)));
+		 (PHYS_RAM_ADDR + (PHYS_RAM_SIZE * 1024U)));
```
Comment (dcpleung) [inline arch/x86/core/x86_mmu.c:1083]: No need to multiply by 1024 anymore.

**Review by dcpleung — CHANGES_REQUESTED** [review]

File: arch/x86/core/x86_mmu.c (hunk 1 above)
Comment (andrewboie) [reply arch/x86/core/x86_mmu.c:24]: If this is already in bytes you need to get rid of the 1024 multiplication to PHYS_RAM_SIZE in the rest of this file!

**Review by tyrantlizard77 — CHANGES_REQUESTED** [review]
This isn't correct, either PHYS_RAM_ADDR is in bytes and all the 1024 multiplications need to be removed, or it is in kilobytes and 1024 must be applied everywhere

**Comment by galak** [comment]
> This isn't correct, either PHYS_RAM_ADDR is in bytes and all the 1024 multiplications need to be removed, or it is in kilobytes and 1024 must be applied everywhere

Its in bytes, and I missed a few cases that @dcpleung pointed out.

File: arch/x86/core/x86_mmu.c (hunk 5 above)
Comment (galak) [reply arch/x86/core/x86_mmu.c]: fixed

File: arch/x86/core/x86_mmu.c (hunk 4 above)
Comment (galak) [reply arch/x86/core/x86_mmu.c]: fixed

File: arch/x86/core/x86_mmu.c (hunk 3 above)
Comment (galak) [reply arch/x86/core/x86_mmu.c]: fixed

File: arch/x86/core/x86_mmu.c (hunk 2 above)
Comment (galak) [reply arch/x86/core/x86_mmu.c]: fixed

**Review by tyrantlizard77 — APPROVED** [review]

**Review by dcpleung — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**